harness/determined/profiler.py

```diff
@@ -91,6 +140,7 @@ def is_enabled(self):
     def update_batch_idx(self, new_batch_idx: int):
         self.current_batch_idx = new_batch_idx
         self.sys_metric_collector_thread.update_batch_idx(self.current_batch_idx)
+        self.send_queue.put(self.timings_batcher.convert_to_post_format())
```

Comment:
Flushing every batch is too many API calls I think. We should be able to handle training at hundreds of batches per second.

Reply:
Yeah, i think i can easily change this to everything N batches, or timing. This was just a rough sketch of the idea and I was kinda avoiding fixing this code much cause I foresaw conflicts.

Reply:
You can prelim review but a lot may change in the next few minutes,  I'm cleaning it up now after rebasing, sorry I'll unassign.

Reply:
No worries, it's good to take a look earlier. 

The main concern I had is that in order to do timing-based flushing, I think we probably need to do it in a separate thread of control - otherwise we have the same issue as the 5 minute timeout. It's not hard, but the more I thought about it, the more I think that TimingBatcher needs its own thread.